spec/fortnox/api/repositories/article_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+require 'fortnox/api'
+require 'fortnox/api/mappers'
+require 'fortnox/api/repositories/customer'
+require 'fortnox/api/repositories/examples/all'
+require 'fortnox/api/repositories/examples/find'
+require 'fortnox/api/repositories/examples/save'
+require 'fortnox/api/repositories/examples/save_with_specially_named_attribute'
+require 'fortnox/api/repositories/examples/search'
+
+describe Fortnox::API::Repository::Article, order: :defined, integration: true do
+  subject(:repository){ described_class.new }
+
+  # include_examples '.save', :description
+
+  include_examples '.save with specially named attribute',
+                   { description: 'Test article' },
+                   :ean,
+                   '5901234123457'
+
+  include_examples '.all', 12
+
+  include_examples '.find', '1' do
+    let( :find_by_hash_failure ){ { description: 'Not Found' } }
+    let( :single_param_find_by_hash ){ { find_hash: { articlenumber: 1 }, matches: 3 } }
```

Comment:
Took me a while to figure out it should be `articlenumber` rather than `article_number`. Wouldn't it be preferable to use the same mappings as for the model itself?

Reply:
```Shell
[1] pry(main)> Fortnox::API::Repository::Article.new.find(article_number: 1)
Fortnox::API::RemoteServerError: Ogiltig parameter i anropet.
from /Users/felix/code/fortnox-api/lib/fortnox/api/request_handling.rb:12:in `raise_api_error'
[2] pry(main)> Fortnox::API::Repository::Article.new.find(articlenumber: 1)
=> [#<Fortnox::API::Model::Article url=nil active=nil article_number="0001" bulky=nil
```

Reply:
Thanks. Something is wrong here... You should be able to use `article_number`.

Reply:
What does the `.yml`-cassette look like?

Reply:
https://github.com/my-codeworks/fortnox-api/pull/113/files#diff-15da9a4e8da28c13ef0e0a4b2e8e85a6R1

Reply:
The exact same thing happens with the customer repo:

```
> Fortnox::API::Repository::Customer.new.find(organisation_number: 1)
Fortnox::API::RemoteServerError: Ogiltig parameter i anropet.
```

Works with `customernumber`.

Reply:
(Personally I would probably like if it was very forgiving so that `customernumber`, `customer_number`, `CustomerNumber` produced the same request. But if you want it more restrictive `customer_number` seems like the obvious choice.)

Reply:
In an effort to keep client code consistent and in keeping with idiomatic Ruby I think we'll only allow for one variant, the underscored one :) But that one should work, so that is a bug...

Reply:
@felixyz & @d-Pixie 
Ok, now I know how this works.

First of all, Fortnox expects the request to look something like this:
```
https://api.fortnox.se/3/articles?articlenumber=1
```

Now to the gem. The `find` method calls `find_all_by` when you send it a hash.
```
        def find_all_by( hash )
          response_hash = get( "#{ self.class::URI }?#{ to_query( hash ) }" )
          instantiate_collection_response( response_hash )
        end
```

There are no use of any mappers or anything here (`to_query` simply turns `articlenumber: 1` to `&articlenumber=1`). Since `find_all_by` simply turns whatever you give it to query parameters, it not only supports attributes, but filters, sorting, searching, limits & offsets as well as pagination. That is why we need to pass `articlenumber` and not `article_number` to the `find`-method. Whether we should change that or not is open for discussion :)

Reply:
@felixyz We'll try to fix this in the future, but for now we need to use `articlenumber`. Thanks for the notice!

Reply:
See #117